components/hab/src/exec.rs

```diff
@@ -60,53 +61,42 @@ pub fn command_from_min_pkg(ui: &mut UI,
                             command: impl Into<PathBuf>,
                             ident: &PackageIdent)
                             -> Result<PathBuf> {
-    try_command_from_min_pkg(ui, command, ident, 0)
-}
-
-fn try_command_from_min_pkg(ui: &mut UI,
-                            command: impl Into<PathBuf>,
-                            ident: &PackageIdent,
-                            retry: u8)
-                            -> Result<PathBuf> {
     let command = command.into();
-    if retry >= RETRY_LIMIT {
-        return Err(Error::ExecCommandNotFound(command));
-    }
-
     let fs_root_path = FS_ROOT_PATH.as_path();
-    match PackageInstall::load_at_least(ident, Some(fs_root_path)) {
-        Ok(pi) => {
-            match fs::find_command_in_pkg(&command, &pi, fs_root_path)? {
-                Some(cmd) => Ok(cmd),
-                None => Err(Error::ExecCommandNotFound(command)),
-            }
-        }
+    let pi = match PackageInstall::load_at_least(ident, Some(fs_root_path)) {
+        Ok(pi) => pi,
         Err(hcore::Error::PackageNotFound(_)) => {
             ui.status(Status::Missing, format!("package for {}", &ident))?;
 
             // JB TODO - Does an auth token need to be plumbed into here?  Not 100% sure.
-            common::command::package::install::start(ui,
-                                                     &default_bldr_url(),
-                                                     &internal_tooling_channel(),
-                                                     &(ident.clone(),
-                                                       PackageTarget::active_target())
-                                                                                      .into(),
-                                                     PRODUCT,
-                                                     VERSION,
-                                                     fs_root_path,
-                                                     &cache_artifact_path(None::<String>),
-                                                     None,
-                                                     // TODO fn: pass through and enable offline
-                                                     // install mode
-                                                     &InstallMode::default(),
-                                                     // TODO (CM): pass through and enable
-                                                     // no-local-package mode
-                                                     &LocalPackageUsage::default(),
-                                                     InstallHookMode::default())?;
-            try_command_from_min_pkg(ui, &command, &ident, retry.saturating_add(1))
```

Comment:
Took me a second to find the key bit you were replacing, and here it is.

This is a great improvement!